### PR TITLE
10493: create validate schemas command

### PR DESCRIPTION
**Pull request by mkalish**
This PR adds the CLI command to validate the schemas in a particular azure blob container

Test Steps:
1. Recompile the prime CLI tool
2. Run `./prime validateSchemas --schema-type="FHIR" --blob-store-connect="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1;QueueEndpoint=http://localhost:10001/devstoreaccount1;" --blob-store-container="metadata"`
3. Verify that the output is `No schemas were found, please make sure that you have the correct configuration`
4. Upload the contents of `/src/test/resources/fhirengine/translation/FHIR_to_FHIR/` using azure storage explorer into the metadata container into the path `fhir_transforms/sender/`
5. Run `./prime validateSchemas --schema-type="FHIR" --blob-store-connect="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1;QueueEndpoint=http://localhost:10001/devstoreaccount1;" --blob-store-container="metadata"`
6. Verify the output is
```
-----
Results from validating FHIR, schemas were valid

1 schemas were validated
-----

Individual Results:

Path validated: http://azurite:10000/devstoreaccount1/metadata/fhir_transforms/sender/simple-transform.yml
Valid: true
Errored while validating: false
```
7. Follow the same steps for HL7

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #10493

### CODE REVIEW

File: prime-router/src/main/kotlin/fhirengine/translation/TranslationSchemaManager.kt [hunk 1]
```diff
@@ -22,8 +23,7 @@ class TranslationSchemaManager : Logging {
 
     companion object {
 
-        private val context = DefaultHapiContext()
-        private val hl7Parser = context.getGenericParser()
+        private val hL7Reader = HL7Reader(ActionLogger())
```
Comment (mkalish) [inline prime-router/src/main/kotlin/fhirengine/translation/TranslationSchemaManager.kt:26]: This change was required due a series of confusions:

- the tests from the initial work that I had forked the last PR did not have assertions that covered whether or not the schemas were valid; they merely checked that validation had occurred
  - This is also why the input files have all changed as they actually weren't valid to begin with
- After the code was actually checking that input matched the output I encountered an issue on the HL7 test where the input/output were identical with the exception that the converted was using `\r` for new lines and the expected output was using `\n` which means they were not equal.  It probably is worth investigating at some point what is causing the difference, but for now I opted to use the same reader class for parsing HL7 strings that the production code uses

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import gov.cdc.prime.router.azure.BlobAccess
+import gov.cdc.prime.router.fhirengine.translation.TranslationSchemaManager
+import org.apache.logging.log4j.kotlin.Logging
+
+class ValidateTranslationSchemaCommand :
+    CliktCommand(
+    name = "validateSchemas",
+    help = "Commands to validate translation schemas"
+),
+    Logging {
```
Comment (jalbinson) [inline prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:16]: This is just our linting rules but very confusing indentation!

```kotlin
class ValidateTranslationSchemaCommand : CliktCommand(
    name = "validateSchemas",
    help = "Commands to validate translation schemas"
), Logging {
```

File: prime-router/src/test/kotlin/cli/ValidateTranslationSchemaCommandTests.kt [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.github.ajalt.clikt.testing.test
+import gov.cdc.prime.router.fhirengine.translation.TranslationSchemaManager
+import io.mockk.every
+import io.mockk.mockkConstructor
+import kotlin.test.Test
+
+class ValidateTranslationSchemaCommandTests {
```
Comment (jalbinson) [inline prime-router/src/test/kotlin/cli/ValidateTranslationSchemaCommandTests.kt:11]: Great example for unit testing future cli commands

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            translationSchemaManager.validateManagedSchemas(
+                schemaType,
+                BlobAccess.BlobContainerMetadata(blobStoreContainer, blobStoreConnection)
+            )
+        if (validationResults.size > 0) {
+            val allPassed = validationResults.all { it.passes }
+
+            val individualResults = validationResults.map { result ->
+                """
+            Path validated: ${result.path}
```
Comment (jalbinson) [inline prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:52]: Another set of confusing indentation. Is this one linter related too?

It's hard to see that this is part of the body of the `map` function at first glance.

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    private val translationSchemaManager = TranslationSchemaManager()
+
+    /**
+     * The schema type to validate
+     */
+    private val schemaType: TranslationSchemaManager.SchemaType by option(
+        "--schema-type",
+        "-s",
+        help = "which schema type to process"
```
Comment (jimmyfagan) [inline prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:26]: (super nit) This help text is formatted differently than the other two, maybe "The schema type to validate against" or maybe just capitalize "which". Or do nothing, this is so unimportant 😆

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.enum
+import gov.cdc.prime.router.azure.BlobAccess
+import gov.cdc.prime.router.fhirengine.translation.TranslationSchemaManager
+import org.apache.logging.log4j.kotlin.Logging
+
+class ValidateTranslationSchemaCommand :
+    CliktCommand(
+    name = "validateSchemas",
+    help = "Commands to validate translation schemas"
```
Comment (jimmyfagan) [inline prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:14]: I could see it being useful for this help text to include more context about the pre-reqs for validation, especially mentioning sample input/output similarly to what is done in the `validateManagedSchemas` function comment.

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt (hunk 5 above)
Comment (jimmyfagan) [reply prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:26]: (less of a nit) This might also be an appropriate place to spell out which folder in the container is used for each schema type. It's visible in the code, but not sure if that's externalized through the CLI at all.

Alternatively, when schemas are not found, the error message could include the schema type directory.

**Review by jimmyfagan — COMMENTED** [review]
Functionally looks good, have a few comments mostly around usability, also had some issues with the test steps in the description. Will huddle with you at some point today, should be able to work through it all pretty quickly, and can approve right after we chat.

**Review by jimmyfagan — APPROVED** [review]
After chatting about it, I'm all good, marking approved!

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     * The schema type to validate
+     */
+    private val schemaType: TranslationSchemaManager.SchemaType by option(
+        "--schema-type",
+        "-s",
+        help = "which schema type to process"
+    ).enum<TranslationSchemaManager.SchemaType>().required()
+
+    private val blobStoreConnection: String by option(
```
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:29]: Could the blob store info be deduced from the environment? Most CLI commands pass in the environment (staging, prod, local) and perform based on that.

**Review by arnejduranovic — APPROVED** [review]

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt (hunk 7 above)
Comment (mkalish) [reply prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:29]: Potentially, though I think it might be limiting to do it that way since this command might need to get run across a couple environments

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt (hunk 4 above)
Comment (mkalish) [reply prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt:52]: Yea, this is how the linter set it up, but I'll see if I can massage it a bit more

File: prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt (hunk 2 above)
Comment (mkalish) [reply prime-router/src/main/kotlin/cli/ValidateTranslationSchemaCommand.kt]: Yep, just how the linter formatted it
